### ISSUE TITLE
Cannot focus space due to an error with the scripting edition

### ISSUE THREAD

**Issue by ando600** [koekeishiya/yabai#1532]
Hi, 
I am new to yabai, and am trying to figure everything out. It seems that I have yabai and skhd setup and everything mostly works. I have followed the instructions regarding adding 
yabai -m signal --add event=dock_did_restart action="sudo yabai --load-sa"
sudo yabai --load-sa

to the top of my .yabairc file. 

When I use shift + alt -5 : yabai -m window --space 5, everything works as I would expect. However, when I try to move to space 5 with yabai -m space --focus 5, I get the error cannot focus due to an error in the scripting edition. 

This is very confusing since I seem to be able to use some scripts but not others. Any help is appreciated. 

Thanks, 

Andrew

**Comment by timhradil** [comment]
This is caused by the script addition injecting being broken on Ventura. This issue has been fixed and merged into master, but not released: https://github.com/koekeishiya/yabai/commit/206c0d6968210f48363f206526760e5750810a61.

You can install from HEAD using these steps from wiki: https://github.com/koekeishiya/yabai/wiki/Installing-yabai-(from-HEAD)

I found my yabai at this path: /opt/homebrew/Cellar/yabai/HEAD-206c0d6/bin/yabai. Rather than at /usr/local/bin/yabai as noted in the wiki.

**Comment by lucafulgenzi** [comment]
I confirm that by installing the HEAD version, everything works correctly.
I also confirm that that the path of yabai is not the one that comes out with the `which yabai` command, but I recommend looking for it manually in the homebrew folders.

**Comment by attiliosbrana** [comment]
I have tried multiple solutions that I've found and I cannot make Yabai --HEAD work, it won't stop asking for Accessibility access.

I created the certificate 'yabai-cert' following the instructions and I signed at the right path:
`codesign -fs 'yabai-cert' /opt/homebrew/Cellar/yabai/HEAD-206c0d6/bin/yabai`

However, it keeps asking me for Accessibility access, and it doesn't work.

**Comment by timhradil** [comment]
I used the $(which yabai) for creating the certificate. I just had to use the /opt/homebrew path for adding yabai to accessibility access manually.

**Comment by ando600** [comment]
I can confirm attiliosbrana's behavior. I have installed yabai from head, and code signed, but when I run brew services start yabai, a popup asking for accessibility access keeps coming up, and even after I specifically point to the location I just code signed in the opt/homebrew/Cellar directory.

**Comment by ando600** [comment]
However when I uninstall and reinstall yabai normally, (i.e. without the --HEAD tag), I do not get this accessibility popup behavior, but the scripting edition error persists.

**Comment by ysomad** [comment]
I have the same issue on Monterey 12.6.1
Disabled SIP, cert is created and signed, yabai from master
Getting `
yabai: scripting-addition failed to inject payload into Dock.app!
unknown key 'topmost'` in the logs.

**Comment by koekeishiya** [comment]
Install the latest version (v5.0.2). Make sure you have followed the wiki instructions exactly as written.
If you still have issues, read this comment for instructions on how to debug your setup: https://github.com/koekeishiya/yabai/issues/1515#issuecomment-1344070256

**Comment by Iddodo** [comment]
This might be a newbie question of sorts, but why am I getting this message and how can it be fixed?


`codesign -fs 'yabai-cert' /opt/homebrew/Cellar/yabai/HEAD-03b2483/bin/yabai
error: The specified item could not be found in the keychain.`

**Comment by behaghel** [comment]
@Iddodo it's definitely a newbie answer but I am stuck here like you with the added challenge of doing it the nix way. I suspect it has to do with the following command returning `0 valid identities found`

    security find-identity -p codesigning -v

I have this would change that (you need to export as .cer `yabai-cert` in Keychain Access)

    sudo security add-trusted-cert -d -r trustRoot /path/to/yabai-cert.cer

This has unblocked me but I suspect nix can't see the keychain I am working on. Let me if it helps.

**Comment by rkshthrmsh** [comment]
Still seeing this issue on Ventura. `yabai-v5.0.2`
- Created `yabai-cert` as per instructions in the wiki
- Used brew to install from HEAD
- Codesigned yabai at:
```
> which yabai
/opt/homebrew/bin/yabai
``` 
- Could NOT find yabai installed in `/usr/local/bin` as per the wiki instructions. Provided accessibility permission to the executables at `/opt/brew/bin/yabai/` and `/opt/brew/Cellar/yabai/HEAD-147b356/bin/yabai` instead.

Issue:
```
> yabai -m space --focus next   
cannot focus space due to an error with the scripting-addition.
```


Is there anything wrong here, or something else that I can try?

**Comment by ando600** [comment]
I am seeing the same behavior as rkshthrmsh

**Comment by jin-ahn** [comment]
any updates? I am on 13.2 and have the issues as @rkshthrmsh

**Comment by olivercarter** [comment]
Same issue as ando600, mac big sur 11.2.2

**Comment by jjpark78** [comment]
Same issue as others, mac ventura 13.3.1 (a)
